contracts/facets/EmbalmerFacet.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.9;
+
+import "hardhat/console.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../libraries/LibTypes.sol";
+import "../libraries/LibEvents.sol";
+import {LibBonds} from "../libraries/LibBonds.sol";
+import {LibUtils} from "../libraries/LibUtils.sol";
+import {AppStorage} from "../storage/LibAppStorage.sol";
+
+contract EmbalmerFacet {
+    AppStorage internal s;
+
+    /// @notice Embalmer creates the skeleton for a new sarcopahgus.
+    /// @param name the name of the sarcophagus
+    /// @param archaeologists the data for the archaeologists
+    /// @param arweaveArchaeologist The address of the archaeologist who uploads to arweave
+    /// @param recipient the address of the recipient
+    /// @param resurrectionTime the resurrection time of the sarcophagus
+    /// @param identifier the identifier of the sarcophagus
+    /// @param sarcoToken The erc20 sarcophagus token
+    /// @param canBeTransfered Whether the sarcophagus can be transferred
+    /// @return The index of the new sarcophagus
+    function initializeSarcophagus(
+        string memory name,
+        LibTypes.Archaeologist[] memory archaeologists,
+        address arweaveArchaeologist,
+        address recipient,
+        uint256 resurrectionTime,
+        bytes32 identifier,
+        IERC20 sarcoToken,
+        bool canBeTransfered
+    ) external returns (uint256) {
+        // Confirm that this exact sarcophagus does not already exist
+        require(
+            !s.sarcophaguses[identifier].exists,
```

Comment:
Wondering if this should use `state` like in v1, b/c after a sarcophagus is cancelled / buried (and potentially cleaned), we would want to represent that state (so we can revert if the sarco is "done")

Reply:
Trying to remember... Did we talk about being able to derive that state somehow? 

If not, then we should make this an enum instead of a boolean.

Reply:
I made this an enum

Reply:
Okay cool, enum should work great. 

The only derived state we should need is whether a sarcophagus can be resurrected (`m` archaeologists have successfully unwrapped). But even then, we would want the state represented by the enum, the recipient would just need to know the sarcophagus can be resurrected by querying the sarcophagus.